contracts/token/ERC20/extensions/ERC4626.sol

```diff
@@ -153,7 +162,7 @@ abstract contract ERC4626 is ERC20, IERC4626 {
         uint256 supply = totalSupply();
         return
             (assets == 0 || supply == 0)
-                ? assets.mulDiv(10**decimals(), 10**_asset.decimals(), rounding)
+                ? assets.mulDiv(10**decimals(), 10**ERC4626.decimals(), rounding)
```

Comment:
To avoid confusion around inheritance, I would define a private function `_assetDecimals()` then use that here and as the default implementation of `decimals()`.